doc/ErlangFormatterComparison.md

```diff
@@ -0,0 +1,345 @@
+See the reasoning behind this decision [here](https://github.com/WhatsApp/erlfmt/blob/master/doc/FormattingDecisionDefaultWidth.md).
+
+## Preserving Representation
+
+During parsing with the default Erlang parser, some information is lost, but since `erlfmt` forked the parser, it can preserve the exact representation of strings, atoms and integers.
+For example:
+
+```erlang
+1_000_000,
+
+"\x61\x62\x63",
+
+"\x61\x62\x63" "\x41\x42\x43",
+
+[1 | [2 | [3 | []]]],
+
+[1 | [2 | [3 | improper_list]]],
+
+{undefined, 'undefined'}.
+```
+
+`rebar3_format` simplifies lists, atoms and strings.
+
+```erlang
+1_000_000,
+
+"\x61\x62\x63",
+
+"abcABC",
+
+[1, 2, 3],
+
+[1, 2, 3 | improper_list],
+
+{undefined, undefined}.
+```
+
+`steamroller` preserves lists, but simplifies strings, atoms and numbers.
+
+```erlang
+1000000,
+"abc",
+"abc" "ABC",
+[1 | [2 | [3 | []]]],
+[1 | [2 | [3 | improper_list]]],
+{undefined, undefined}.
+```
+
+`erl_tidy` simplifies all the examples.
+
+```erlang
+1000000,
+
+"abc",
+
+"abcABC",
+
+[1, 2, 3],
+
+[1, 2, 3 | improper_list],
+
+{undefined, undefined}.
+```
+
+## Line Break Hints
+
+`erlfmt` tries to fit as much as it can on a single line, but will respect user introduced line breaks hints and empty lines in certain places and preserve them.
+For example in lists, where the whole list is broken up, if it contains a line break.
+
+```erlang formatted linehints
+-export([
+    %% public functions
+    foo/1, foo/2,
+    bar/1,
+
+    %% testing helpers
+    baz/5
+]).
+
+x() ->
+    Foo = [
+        short,
+        list
+    ].
+```
+
+`rebar3_format` loses all new lines and moves the comments out of the list, since the list can fit on a single line, completely loosing the information about grouping.
+
+```erlang
+-export([foo/1, foo/2, bar/1, baz/5]).    %% public functions
+
+    %% testing helpers
+
+x() ->
+    Foo = [short, list].
+```
+
+`steamroller` preserves the newlines in the export grouping, since there were comments that forced newlines to exist inside the list, but does not preserve newlines in the short list.
+
+```erlang
+-export(
+  [
+    %% public functions
+    foo/1,
+    foo/2,
+    bar/1,
+    %% testing helpers
+    baz/5
+  ]
+).
+
+x() -> Foo = [short, list].
+```
+
+`erl_tidy` kept some new lines inside the export statement, we assume because of it originally containing comments.
+
+```erlang
+-export([bar/1,
+         baz/5,
+         foo/1,
+         foo/2]).    %% public functions
+
+    %% testing helpers
+
+x() -> Foo = [short, list].
+```
+
+If we reformat with `erl_tidy` again these newlines are also removed, because as expected the comments were no longer inside the export statement.
+
+```erlang
+-export([bar/1, baz/5, foo/1, foo/2]).
+
+                   %% public functions
+
+    %% testing helpers
+
+x() -> Foo = [short, list].
+```
+
+## Opt In/Out
+
+`erlfmt` is the only Erlang formatter, as far as we know, that allows you to opt-in per file and opt-out per top-level expression.
```

Comment:
`rebar3_format` allows you to opt out from formatting, by adding `-format ignore.` in files, or by adding an `ignore` configuration option either in `rebar.config` or when running `rebar3 format` in the command line with a blob of the files you want not to format.
That applies regardless of the formatter you choose. If you happen to use `erlfmt` within `rebar3_format`, you can also use `@format`, of course.

Reply:
> Some clarifications regarding `rebar3_format`.
> 
> At some point I think it would also be worth noting that wherever you mention `rebar3_format` in your text, you're actually referring to the `default_formatter` within `rebar3_format`. You can still use any other formatter with `rebar3_format` if you want to (including, of course, `erlfmt`).

Very good point.  I have added a clarifying sentence to the top, right below the table.